### DISCUSSION TITLE
New vase mode bins! Testing needed

### DISCUSSION THREAD

**Discussion by kennetek** [kennetek/gridfinity-rebuilt-openscad#6] [Announcements]
Tried my hand at adapting the bins to be vase mode compatible. It works with my slicer settings, but given how it will probably be finicky based on the printer, it would be really helpful if you all could try it out and make sure it slices correctly. Along with catching any other mistakes I made, of course. Would love to get feedback. Thanks!

**Comment by unteins** [comment]
I'm sort of testing this out - I will print some actual base mode bins in a bit.

However, what I am actually doing is abusing this to create multi-part prints to fit huge bins on a smaller printer. I need some 8x2 bins and I can't fit them on a Prusa. 

Would love to see an option to break a bin down into shorter sections that can go together (JointSCAD Library might help: https://github.com/HopefulLlama/JointSCAD) - maybe simple finger joints at the break so there's a decent amount of surface area. Was thinking that integrated supports for the joints would be good. 

Anyway, to the topic at hand, one thing that is WAY nicer in the Vase Mode is the settings for customizing - mostly clear and easy to try out. I didn't notice that some settings changes don't re-render the model though.

Last thing is that it was not at all clear how to get the BASE for the printed vase mode box. I eventually figured something out, but it might be useful to add a toggle for "generate base" so that the output is ready to go with the right number of bases (per the number of spots) included in the output.

**Comment by kennetek** [comment]
Yeah, I figured out the customizer was a thing between writing the rebuilt script and writing the vase script. However, the rebuilt script was recently updated to have better customizer formatting, so that should be good now. 

As for the bases, I did think about the best way to make it clear how to use them. Ideally, the bases should be printed together using a "complete individual objects" option in your slicer, and then the main bin separately. I wanted to include a STL file for the base inside the folder, but the base dimensions change with your print settings. Generating the bin and the base next to each other seems like it encroaches on the responsibility of the slicer (arrangement of parts). I'll try to improve the instructions to make it clear how to get the base (calling `gridfinityBaseVase()`).

**Comment by unteins** [comment]
When I called that function I only got a single base piece. But good to know it changes with the other settings.

**Reply by unteins** [reply]
I don't know if there is a way for OpenSCAD to generate 2 different pieces - but most slicers can split apart unconnected pieces. Of course, if you export as 3DMF I think you get individual parts too and can rearrange in the slicer. Though I would say plenty of models are uploaded with the pieces pre-arranged in optimal orientation for a certain size build plate. I don't know if there's a canonically correct option though.

**Reply by kennetek** [reply]
The function only generates a single base piece. That's intentional - because then when you go over to your slicer, you just make as many copies as you need to fill the X-shaped cutouts on the bottom of the bin. Then, you don't have to generate a new base STL file each time you make a new bin (assuming you use the same nozzle and layer heights each time), just make the amount of necessary copies in your slicer.

**Reply by unteins** [reply]
Yeah, looking at it, I see how it works now - in order to make the base pieces for vase modes it would be very hard to have multiple bases connected together.

**Comment by CullenJWebb** [comment]
This is so impressive. I'll be testing this out (I use SuperSlicer) and I'll report back on the results.

I have been printing several 3x5x6 bins which consume about 200g each even with custom tweaks to reduce filament where possible so this could really save my bacon.

**Comment by CullenJWebb** [comment]
I haven't printed anything yet, but the "enable pinch" toggle reminded me that ribbing can greatly increase durability and reduce warping: https://youtu.be/3-ygdNQThAs?t=437

If a rib pattern could be added to the flat walls I think that would help a lot, but I will let you know if I still think that is necessary after I print a few.

**Reply by kennetek** [reply]
I agree, I have seen that video. But like, effort, you know? Doing that kind of stuff in OpenSCAD would be pain. I made a few test bins in Solidworks because I was interested as well, and what I found was that the two walls seemed to give the bins enough strength, at least compared to the ridged single-walled test pieces. That's one of the "limitations" of Stefan's video I guess, it only addresses single walls. I suppose that means ridged two-walled parts would be basically indestructible, but then it comes back to trying to model the ridges in OpenSCAD. 

But yeah, the pinch option basically does that. But it solves a different issue, without that pinch, the upper part of the lip is hollow, with only a single layer holding the walls together at the top. And because its the top layer, it does not actually go completely around the part due to the nature of vase mode. So what would happen is that the walls would separate at the top, and the bin's strength would drastically reduce.

**Reply by CullenJWebb** [reply]
Yeah I'm just putzing around in OpenSCAD, I'm a beginner and it looks like it would be very difficult to implement. The more I think about it the more I think it would not be worth the effort.

**Comment by CullenJWebb** [comment]
I am having terrific success with the vase mode, it is now my primary form of Gridfinity bins. However, I noticed 2 things which I wanted to share:

### Funnel glitch when gridz < 5

![funnel-debug](https://user-images.githubusercontent.com/2952787/204593473-cd5e9b41-61fa-4c56-a0c6-c38cfd30c92e.PNG)

This error only appears at gridz=4 and lower. It's easy enough to disable but FYI.

### Suggestion: Alternate seam position

![20221129_115422](https://user-images.githubusercontent.com/2952787/204595047-b423cc3e-f890-4ae9-9ae8-42b8ac41feb9.jpg)

This is a small nitpick but I noticed that the dividers alternate direction on each layer between the 2 sides of the bin, resulting in a seamless and durable bond. The seam along the outer wall however is a consistent vertical line and while a minor issue I think that the durability and appearance of the bins would be improved if the seam were to either (A) alternate a few mm back and forth along the same wall, much like an alternating brick pattern, or (B) alternate back and forth between the 2 opposite walls.

This is coming from a absolute beginner so I have no idea if what I'm suggesting is feasible. I'm going to happily continue using and recommending these bins regardless, just wanted to share some feedback.

Thank you so much for creating this tool. I hope that you sleep a smidgen easier at night knowing that some workshop somewhere on the planet is a little tidier now thanks to this repo!

**Reply by kennetek** [reply]
Thank you for pointing out the vase mode funnel error. I'll fix that soon. As for the seam, it could definitely be improved. This was the easiest way of doing it, just making a thin vertical slice. Since I've already got code to alternate between layers, it might work, but who knows what the slicer would do. I'll definitely look into it.

**Comment by misterpok** [comment]
I was having trouble with encouraging Cura to slice the vase modes the way I wanted. 
![image](https://user-images.githubusercontent.com/5810793/208219082-b97be2ba-43b2-49da-b2c4-3b5ff7b90f58.png)
Firstly, the bottom section would not print the inner wall, resulting in no dividers there.
Disabling 'Remove all holes' resulted in this:
![image](https://user-images.githubusercontent.com/5810793/208219274-2c4eaa5b-e39e-4d96-ad7a-5e311d9252ac.png)
which was a step in the right direction.
Changing 'surface mode' to 'surface' resulted in good walls, but no floor. I suspect that Cura was not respecting vase mode here, though. However, with a different way of mounting to the baseplate, this could be a workaround.
Setting the slicing tolerance to 'exclusive' and disabling 'remove all holes' has resulted in the best preview slice so far. Leading me to think this is a combination of Cura removing the seam and not slicing evenly, or at some offset.
![image](https://user-images.githubusercontent.com/5810793/208219903-166461dd-d0ac-4406-9f8f-15b7b9c0937c.png)
Those travel moves are *odd*.

**Comment by greenbigfrog** [comment]
Really nice, but I just seem to be unable to get this stuff to slice correctly using superslicer... Not sure how I can give more info, but no matter what I try, it just won't slice right. I think I'll have to sit down and try again from scratch, and also document my attempts here, so that I can actually possibly receive help. /rant over, Sorry

**Reply by greenbigfrog** [reply]
Ok, actually might have sorted this out, maybe. I had the issue of there being seams and travel moves no matter what I did.
But looking at the stl, there's no slit or anything, that would make it a true vase. So there will always be some travel/seam.
![](https://i-got.ratelimited.today/U9u7Pcr.png)
My main issue was having the flow & width settings in superslicer not match up what I had the in the scad. Set default width to same as in the scad, and all other extrusion width to 0.

But perhaps my understanding here is completely wrong?

**Reply by greenbigfrog** [reply]
Looking at the gcode uploaded at https://www.printables.com/model/284371-spiral-vase-gridfinity-in-openscad/files I think it's intentional.

**Comment by saywebsolutions** [comment]
I can't seem to to get cura to slice the bottom x cutouts of the bins; instead it just prints a flat/solid bottom

**Reply by CullenJWebb** [reply]
Is your brim by change being printed inside the x?

**Reply by saywebsolutions** [reply]
I don't believe I am using a brim
